### PR TITLE
Mongodb item writer implementation

### PR DESCRIPTION
**Pull request by 108krohan**
### Revision 2
- Assembly checker processor

#### Test

- Sample document: 
```
{ 
	"_id" : "13CB23F25B5A21C0D13B839F43A31CD50BD49414", 
	"_class" : "uk.ac.ebi.eva.accession.dbsnp2.model.ClusteredVariantEntity", 
	"referenceAllele" : "TG", 
	"assemblyMatch" : false, 
	"asm" : "GCF_000001405.38", 
	"tax" : 9606, 
	"contig" : "NC_000024.10", 
	"start" : NumberLong(5793498), 
	"type" : "MNV", 
	"accession" : NumberLong(34072347), 
	"version" : 1, 
	"createdDate" : ISODate("2006-03-13T11:23:00Z") 
}
```

### Revision 1
- Writer implementation

#### Test
- Collection updates with 25 documents in it. Rerun of job results in same 25 documents in collection.
Sample document:
```
{
	 "_id" : "13CB23F25B5A21C0D13B839F43A31CD50BD49414", 
	"_class" : "uk.ac.ebi.eva.accession.core.persistence.DbsnpClusteredVariantEntity", 
	"asm" : "GCF_000001405.38", 
	"tax" : 9606, 
	"contig" : "NC_000024.10", 
	"start" : NumberLong(5793498), 
	"type" : "MNV", 
	"accession" : NumberLong(34072347), 
	"version" : 1, 
	"createdDate" : ISODate("2006-03-13T11:23:00Z") 
}
```

### CODE REVIEW

File: eva-accession-import-dbsnp2/src/main/java/uk/ac/ebi/eva/accession/dbsnp2/configuration/ImportDbsnpJsonVariantsWriterConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+import static uk.ac.ebi.eva.accession.dbsnp2.configuration.BeanNames.DBSNP_JSON_VARIANT_WRITER;
+
+@Configuration
+public class ImportDbsnpJsonVariantsWriterConfiguration {
+
+    private static final Logger logger = LoggerFactory.getLogger(ImportDbsnpJsonVariantsWriterConfiguration.class);
+
+    @Bean(name = DBSNP_JSON_VARIANT_WRITER)
+    @StepScope
+    public MongoItemWriter<ClusteredVariant> writer(MongoTemplate mongoTemplate) {
```
Comment (cyenyxe) [inline eva-accession-import-dbsnp2/src/main/java/uk/ac/ebi/eva/accession/dbsnp2/configuration/ImportDbsnpJsonVariantsWriterConfiguration.java]: Why not using [the DbsnpClusteredVariantWriter class](https://github.com/EBIvariation/eva-accession/blob/master/eva-accession-import/src/main/java/uk/ac/ebi/eva/accession/dbsnp/io/DbsnpClusteredVariantWriter.java)? You can always convert from `ClusteredVariant` to `DbsnpClusteredVariantEntity` beforehand.

File: eva-accession-import-dbsnp2/src/main/java/uk/ac/ebi/eva/accession/dbsnp2/io/JsonNodeToClusteredVariantProcessor.java [hunk 2]
```diff
@@ -34,11 +34,12 @@
 /**
  * Processes a DbSNP JsonNode to produce a clustered variant entity.
  */
-public class JsonNodeToClusteredVariantProcessor implements ItemProcessor<JsonNode, DbsnpClusteredVariantEntity> {
+public class JsonNodeToClusteredVariantProcessor implements ItemProcessor<JsonNode, ClusteredVariantEntity> {
```
Comment (cyenyxe) [inline eva-accession-import-dbsnp2/src/main/java/uk/ac/ebi/eva/accession/dbsnp2/io/JsonNodeToClusteredVariantProcessor.java]: We are importing variants from dbSNP, so this should still be a `DbsnpClusteredVariantEntity`.

File: eva-accession-import-dbsnp2/src/main/java/uk/ac/ebi/eva/accession/dbsnp2/configuration/ImportDbsnpJsonVariantsWriterConfiguration.java (hunk 1 above)
Comment (108krohan) [reply eva-accession-import-dbsnp2/src/main/java/uk/ac/ebi/eva/accession/dbsnp2/configuration/ImportDbsnpJsonVariantsWriterConfiguration.java]: I thought only classes from `commons` and `core` packages could be used

Comment (108krohan) [reply eva-accession-import-dbsnp2/src/main/java/uk/ac/ebi/eva/accession/dbsnp2/configuration/ImportDbsnpJsonVariantsWriterConfiguration.java]: This is in the `eva-accession-import` module. So I thought to not reuse it, but I'll correct this in a new PR if it's applicable

**Pull request closed without merging**